File: src/contests/cg/puzzle/objectivec/the-fastest.m
// 🎮 CodinGame Puzzle - the-fastest
// https://www.codingame.com/training/medium/the-fastest

#import <Foundation/Foundation.h>
#include <string.h>

int main() {
    // Times are HH:MM:SS so lexicographic comparison equals chronological comparison.
    int n;
    scanf("%d", &n);
    char best[64];
    best[0] = '\0';
    char t[64];
    for (int i = 0; i < n; i++) {
        scanf("%63s", t);
        if (best[0] == '\0' || strcmp(t, best) < 0) strcpy(best, t);
    }
    printf("%s\n", best);
    return 0;
}

File: src/contests/cg/puzzle/vbnet/the-fastest.vb
' 🎮 CodinGame Puzzle - the-fastest
' https://www.codingame.com/training/medium/the-fastest

Imports System

Module Solution
    Sub Main()
        ' Times are HH:MM:SS so lexicographic comparison equals chronological comparison.
        Dim n As Integer = Integer.Parse(Console.ReadLine().Trim())
        Dim best As String = ""
        For i As Integer = 0 To n - 1
            Dim t As String = Console.ReadLine().Trim()
            If best = "" OrElse String.CompareOrdinal(t, best) < 0 Then
                best = t
            End If
        Next
        Console.WriteLine(best)
    End Sub
End Module
